python/kernmlops/kernmlops_benchmark/mongodb.py

```diff
@@ -0,0 +1,99 @@
+import subprocess
+from dataclasses import dataclass
+from typing import cast
+
+from data_schema import GraphEngine, demote
+from kernmlops_benchmark.benchmark import Benchmark, GenericBenchmarkConfig
+from kernmlops_benchmark.errors import (
+  BenchmarkNotInCollectionData,
+  BenchmarkNotRunningError,
+  BenchmarkRunningError,
+)
+from kernmlops_config import ConfigBase
+
+
+@dataclass(frozen=True)
+class MongoDbConfig(ConfigBase):
+  record_count: int = 1000000
+  readProportion: float = 0.25
+  updateProportion: float = 0.75
+
+
+class MongoDbBenchmark(Benchmark):
+
+    @classmethod
+    def name(cls) -> str:
+        return "mongodb"
+
+    @classmethod
+    def default_config(cls) -> ConfigBase:
+        return MongoDbConfig()
+
+    @classmethod
+    def from_config(cls, config: ConfigBase) -> "Benchmark":
+        generic_config = cast(GenericBenchmarkConfig, getattr(config, "generic"))
+        mongodb_config = cast(MongoDbConfig, getattr(config, cls.name()))
+        return MongoDbBenchmark(generic_config=generic_config, config=mongodb_config)
+
+    def __init__(self, *, generic_config: GenericBenchmarkConfig, config: MongoDbConfig):
+        self.generic_config = generic_config
+        self.config = config
+        self.benchmark_dir = self.generic_config.get_benchmark_dir() / "ycsb"
+        self.process: subprocess.Popen | None = None
+
+    def is_configured(self) -> bool:
+        return True
+        # print(f'is_configured directory name: {self.benchmark_dir}')
+        return self.benchmark_dir.is_dir()
+
+    def setup(self) -> None:
+        if self.process is not None:
+            raise BenchmarkRunningError()
+        self.generic_config.generic_setup()
+
+    def run(self) -> None:
+        if self.process is not None:
+            raise BenchmarkRunningError()
+
+        self.process = subprocess.Popen(
+            [
+                f"{self.benchmark_dir}/ycsb-0.17.0/bin/ycsb",
+                "run",
+                "mongodb",
+                "-s",
+                "-P",
+                f"{self.benchmark_dir}/ycsb-0.17.0/workloads/workloada",
+                "-p",
+                f"operationcount={self.config.record_count}",
```

Comment:
please change the config argument from `record_count` to `operation_count` to reflect how it is used